File: backend/Models/Empleado.cs
using System.ComponentModel.DataAnnotations;

namespace backend.Models
{
    public class Empleado
    {
        [Key]
        [Required]
        [StringLength(20)]
        public string Codigo { get; set; } = string.Empty;
        [Required]
        [StringLength(150)]
        public string Nombres { get; set; } = string.Empty;
        [StringLength(11)]
        public string? TipoDocumento { get; set; }
        [StringLength(20)]
        public string? Cedula { get; set; }
        [Required]
        [StringLength(15)]
        public string EStatus { get; set; } = "ACTIVO";
        [StringLength(100)]
        public string? Puesto { get; set; }
        public DateTime? FechaIngreso { get; set; }
        public DateTime? FechaNacimiento { get; set; }
        [EmailAddress]
        [StringLength(150)]
        public string? Email { get; set; }
         public DateTime FechaCreacion { get; set; } = DateTime.UtcNow;
        public DateTime FechaActualizacion { get; set; } = DateTime.UtcNow;
    }
}

File: sistema-nomina/src/types/empleado.ts
export interface Empleado {
  codigo: string;
  nombres: string;
  tipoDocumento?: string | null;
  cedula?: string | null;
  eStatus: string;
  puesto?: string | null;
  sueldoBase?: number | null;
  fechaIngreso?: string | null;
  fechaNacimiento?: string | null;
  email?: string | null;
  fechaCreacion?: string;
  fechaActualizacion?: string;
}